### ISSUE TITLE
Path double hex encoded at assign

### ISSUE THREAD

**Issue by markov2** [sysread/URI-Fast#46]
When I fix the path, because html_url() does not handle utf8 input, I use something like this:
```perl
$url->path(do_something_smart($url->path));
```
My smart function translates UTF8 into HEX encoding.  However, when I later look in $url->path, all the "%" signs have become "%25".

Is it possible that the setter for path() expects unencoded strings, and the path getter produces encoded strings?  It is really very inconvenient design when `$url->path($url->path)` is not idempotent.  And what is my way around setting path without mutilation?

**Comment by sysread** [comment]
Does `raw_path` work for that case?

**Comment by sysread** [comment]
We have "raw" getters; I can make an analogous setter that would allow you to directly set the path attribute without encoding the value. Then you could do something like:

```
use URI::Fast qw(uri encode);

my $uri = uri(...);
my $path = encode(..., "/"); # encode everything but "/"
$uri->raw_path($path);
```

**Comment by sysread** [comment]
Ok, I added this in de97825. Would you mind taking a look and seeing if that would handle your use case sufficiently?

**Comment by sysread** [comment]
I have a dev release with this on cpan. I think I'll go ahead and promote it to full release and close this issue. Feel free to reopen if this does not cover the use case.

**Comment by sysread** [comment]
Fixed in v0.55, now on CPAN.
